stdlib/public/core/Collection.swift

```diff
@@ -56,8 +57,66 @@ public protocol Indexable {
   ///
   /// - Complexity: O(1)
   subscript(position: Index) -> _Element { get }
+  
+// TODO: swift-3-indexing-model - replaces functionality in ForwardIndex
+  /// Performs a range check in O(1), or a no-op when a range check is not
+  /// implementable in O(1).
+  ///
+  /// The range check, if performed, is equivalent to:
+  ///
+  ///     precondition(bounds.contains(index))
+  ///
+  /// Use this function to perform a cheap range check for QoI purposes when
+  /// memory safety is not a concern.  Do not rely on this range check for
+  /// memory safety.
+  ///
+  /// The default implementation for forward and bidirectional indices is a
+  /// no-op.  The default implementation for random access indices performs a
+  /// range check.
+  ///
+  /// - Complexity: O(1).
+  func _failEarlyRangeCheck(index: Index, bounds: Range<Index>)
+  
+// TODO: swift-3-indexing-model - replaces functionality in ForwardIndex
+  /// Performs a range check in O(1), or a no-op when a range check is not
+  /// implementable in O(1).
+  ///
+  /// The range check, if performed, is equivalent to:
+  ///
+  ///     precondition(
+  ///       bounds.contains(range.startIndex) ||
+  ///       range.startIndex == bounds.endIndex)
+  ///     precondition(
+  ///       bounds.contains(range.endIndex) ||
+  ///       range.endIndex == bounds.endIndex)
+  ///
+  /// Use this function to perform a cheap range check for QoI purposes when
+  /// memory safety is not a concern.  Do not rely on this range check for
+  /// memory safety.
+  ///
+  /// The default implementation for forward and bidirectional indices is a
+  /// no-op.  The default implementation for random access indices performs a
+  /// range check.
+  ///
+  /// - Complexity: O(1).
+  func _failEarlyRangeCheck(rangeStart rangeStart: Index,
+                            rangeEnd: Index,
+                            boundsStart: Index,
+                            boundsEnd: Index)
```

Comment:
on Tue Feb 23 2016, Dmitri Gribenko <notifications-AT-github.com> wrote:

> Please don't indent code like that, aligning it to an arbitrary point on the previous line.  If you need to break this line, please do it like this:
>
> ```swift
>   func _failEarlyRangeCheck(
>     rangeStart rangeStart: Index,
>     rangeEnd: Index,
>     boundsStart: Index,
>     boundsEnd: Index)
> ```

FWIW, every parameter needn't have its own line:

```swift
  func _failEarlyRangeCheck(
    rangeStart rangeStart: Index, rangeEnd: Index,
    boundsStart: Index, boundsEnd: Index)
```

is fine too, especially in this case where logical groupings can be
made.

-- 
-Dave